colorutils.cpp

```diff
@@ -443,6 +443,69 @@ CRGB ColorFromPalette( const CRGBPalette16& pal, uint8_t index, uint8_t brightne
     return CRGB( red1, green1, blue1);  
 }
 
+CRGB ColorFromPaletteExtended( const CRGBPalette16& pal, uint16_t index, uint8_t brightness, TBlendType blendType)
+{
+    // This fucnction has the same intuitive behavior as the other ColorFromPalette
+    // functions, except it provides 8-bit interpolation between palette entries.
+
+    // Extract the four most significant bits of the index as a palette index.
+    uint8_t index_4bit = (index >> 12);
+
+    // Calculate the 8-bit offset from the palette index.
+    // Throws away the 4 least significant bits and uses the middle 8.
+    uint8_t offset = (uint8_t)(index >> 4);
+
+    // Get the palette entry from the 4-bit index
+    //  CRGB rgb1 = pal[ hi4];
+    const CRGB* entry = &(pal[0]) + index_4bit;
+    uint8_t red1   = entry->red;
+    uint8_t green1 = entry->green;
+    uint8_t blue1  = entry->blue;
+
+    uint8_t blend = offset && (blendType != NOBLEND);
+
+    if( blend ) {
+        // If palette blending is enabled, use the offset to interpolate between
+        // the selected palette entry and the next.
+
+        if( index_4bit == 15 ) {
+            entry = &(pal[0]);
+        } else {
+            entry++;
+        }
+
+        // Calculate the scaling factor and scaled values for the lower palette value.
+
+        uint8_t f1 = 256 - offset;
```

Comment:
I think the problem described in #1014 has to do with this line. All other ColorFromPalette's calc f1 like this:
`uint8_t f1 = 255 - f2;`